src/components/UserSelector.tsx

```diff
@@ -1,19 +1,47 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { User } from '../types/User';
+import { Post } from '../types/Post';
+
+type Props = {
+  users: User[],
+  selectedUser: User | null,
+  setSelectedUser: React.Dispatch<React.SetStateAction<User | null>>,
+  setSelectedPost: React.Dispatch<React.SetStateAction<Post | null>>,
+};
+
+export const UserSelector: React.FC<Props> = React.memo(({
+  users,
+  selectedUser,
+  setSelectedUser,
+  setSelectedPost,
+}) => {
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+
+  const handleUserChange = (user: User) => {
+    setSelectedUser(user);
+    setIsDropdownActive(false);
+    setSelectedPost(null);
+  };
 
-export const UserSelector: React.FC = () => {
   return (
     <div
       data-cy="UserSelector"
-      className="dropdown is-active"
+      className={cn('dropdown', {
+        'is-active': isDropdownActive,
+      })}
     >
       <div className="dropdown-trigger">
         <button
           type="button"
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setIsDropdownActive(true)}
```

Comment:
The dropdown opens on click, but there's currently no way to close it without selecting a user. Consider adding a way to close the dropdown by clicking outside the component or pressing the ESC key.